src/app/student-note-dialog/student-note-dialog.component.ts

```diff
@@ -0,0 +1,101 @@
+import {Component, Inject, OnInit} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { Note, Rank, Student } from 'src/types/student';
+import { SpreadsheetService } from '../spreadsheet.service';
+import { firstValueFrom, forkJoin, pipe, timeout } from 'rxjs';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { handleError } from '../util/error-util';
+import { shouldTrackRR } from '../util/rank-util';
+
+@Component({
+    selector: 'dialog-student-note',
+    templateUrl: './student-note-dialog.component.html',
+    styleUrls: ['./student-note-dialog.component.scss'],
+})
+export class StudentNoteDialogComponent implements OnInit {
+    loading: boolean = false;
+    noteForm: FormGroup;
+    student: Student;
+    note: Note | null = null;
+
+    shouldTrackRR = shouldTrackRR; // Required for reference in html
```

Comment:
This was super confusing for me to read at first - I couldn't exactly figure out what you meant. Honestly just updating the comment to "exporting library function for use in HTML" or something would've made me realize sooner that this was something you had to import and forward outward